cmd/controller/app/controller.go

```diff
@@ -177,3 +182,9 @@ func KubeConfig(apiServerHost string) (*rest.Config, error) {
 
 	return cfg, nil
 }
+
+func NamespaceFilter(namespace string) func(listOptions *metav1.ListOptions) string {
```

Comment:
❓ Could you use type `FilterFunc` for the return signature? Is `FilterFunc` public? If not, should it be?

Reply:
The issue is that the `FilterFunc` type is generated within each `informer` package, meaning that we'd need to explicitly cast between the two types when we call `NewSharedInformerFactory`.

Ultimately it's a limitation in Go's type system - it doesn't inspect the 'value' side of the type when assessing whether something in fact implements the type, so two types that look identical to you or I, are different according to Go (although *can* be cast, which *does* take account of the 'value' side). There's a good blog post on this somewhere, but can't find it right now. Ping me if you want me to keep digging for it :smile: